Gem/Settings/ViewModels/ChainNodeViewModel.swift

```diff
@@ -2,23 +2,93 @@
 
 import Foundation
 import Primitives
+import BigInt
+import Components
+import Style
 
 struct ChainNodeViewModel {
     let chainNode: ChainNode
-    
+    let nodeMetrics: NodeMetrics?
+    let valueFormatter: ValueFormatter
+
+    init(chainNode: ChainNode, nodeMetrics: NodeMetrics?, valueFormatter: ValueFormatter) {
+        self.chainNode = chainNode
+        self.nodeMetrics = nodeMetrics
+        self.valueFormatter = valueFormatter
+    }
+
     var title: String {
-        guard let host = chainNode.host else {
-            return ""
-        }
-        if host.contains("gemnodes.com") {
-            return "Gem Wallet Node"
-        }
-        return host
+        guard let host = chainNode.host else { return "" }
+        return chainNode.isGemNode ? "Gem Wallet Node" : host
+    }
+
+    var titleExtra: String? {
+        NodeMetricsFormatter(metrics: nodeMetrics)
+            .latestBlockFormatted(
+                latestBlockTitle: Localized.Nodes.ImportNode.latestBlock,
+                valueFormatter: valueFormatter
+            )
+    }
+
+    var subtitle: String? {
+        NodeMetricsFormatter(metrics: nodeMetrics)
+            .latencyFormatted
     }
+
+    var placeholders: [ListItemViewPlaceholderType] {
+        nodeMetrics?.error != nil ? [] : [.subtitle]
+    }
+}
+
+// MARK: - Identifiable
+
+extension ChainNodeViewModel: Identifiable {
+    var id: String { chainNode.id }
 }
 
+// MARK: - Models extensions
+
 extension ChainNode {
     var host: String? {
-        return URL(string: node.url)?.host
+        URL(string: node.url)?.host
+    }
+
+    var isGemNode: Bool {
+        host?.contains("gemnodes.com") ?? false
+    }
+}
+
+// MARK: - Formatter
+
+struct NodeMetricsFormatter {
+    let metrics: NodeMetrics?
+
+    func latestBlockFormatted(latestBlockTitle: String, valueFormatter: ValueFormatter) -> String? {
```

Comment:
```suggestion
    func latestBlockText(latestBlockTitle: String, valueFormatter: ValueFormatter) -> String? {
```

To simplify, just use `text` if it returns formatted data